eradiate/solvers/onedim/app.py

```diff
@@ -302,75 +496,79 @@ def _configure_scene(self):
-                    config["surface"]["width"] = atmosphere.kernel_width
-
-                self._elements["surface"] = SurfaceFactory.create(config["surface"])
-
-                # Set measure
-                for config_measure in self.config["measure"]:
-                    if atmosphere is not None:
-                        height = atmosphere.kernel_height.magnitude
-                        offset = atmosphere.kernel_offset.magnitude
-                    else:
-                        height = 0.1
-                        offset = 0.001
-                    config_measure["origin"] = [0, 0, height + offset]
-
-                    if config_measure["type"] in self._measure_aliases["toa_pplane"]:
-                        if "orientation" not in config_measure:
-                            # TODO: fix this behaviour (will crash if illumination.type is not directional)
-                            #  Suggested change: raise if no orientation and illumination.type is not directional
-                            phi_i = self._elements["illumination"].azimuth.to(ureg.rad).magnitude
-                            config_measure["orientation"] = [np.cos(phi_i), np.sin(phi_i), 0]
-
-                        if not config_measure.get("id", None):
-                            config_measure["id"] = config_measure["type"]
-                        config_measure["type"] = "radiancemeter_pplane"
-
-                    elif config_measure["type"] in self._measure_aliases["toa_hsphere"]:
-                        if not config_measure.get("id", None):
-                            config_measure["id"] = config_measure["type"]
-                        config_measure["type"] = "radiancemeter_hsphere"
-
-                    else:
-                        raise ValueError(f"unsupported measure type '{config_measure['type']}'")
-
-                    config_measure["hemisphere"] = "back"
-                    # TODO: warn when overriding parameters set by user
-
-                    self._elements[config_measure["id"]] = \
-                        MeasureFactory.create(config_measure)
-                    sensor_info = self._elements[config_measure["id"]].sensor_info()
-                    for sensor_id, sensor_spp in sensor_info:
-                        self._measure_registry.insert(
-                            {
-                                "measure_id": config_measure["id"],
-                                "sensor_id": sensor_id,
-                                "sensor_spp": sensor_spp
-                            }
-                        )
-
-                # Expand elements to kernel scene dictionary
-                self._kernel_dict.add(list(self._elements.values()))
+                    offset = atmosphere.kernel_offset.to(cdu.get("length")).magnitude
+                    height = atmosphere.kernel_height.to(cdu.get("length")).magnitude - offset
+
+                else:
+                    height = 0.1
+                    offset = 1e-3
```

Comment:
If `atmosphere` is `None`, I assume that `height` defines the top-of-atmosphere altitude where the sensor is placed for a `toa` measure, right? Why is`offset` not 0 though? There will not be any intersections issues without an atmosphere shape, if I understand correctly.

Reply:
That's a good question. I actually removed this use case later on, but maybe it should be reinstated? I'll think about that.